README.md
# CMPE295B
SJSU


history
curl -k -X POST -d "{\"username\": \"vishnu\", \"email\": \"vishnu@cisco.com\", \"password\": \"vishnu123\"}" https://localhost:8443/user
  
curl -k -X PUT -d "{\"billingContact\": \"Vishnu Konepalli\", \"email\": \"vishnu@cisco.com\", \"address\": \"3550 Cisco Way San Jose CA 95134\", \"zipCode\": \"95134\", \"carLicensePlat\": \"5jdg099\"}" https://localhost:8443/user/vishnu/profile
  
curl -k -X POST -d "{\"Parkingid\": \"A123\"}" https://localhost:8443/user/vishnu/park
  
curl -k -X PUT -d "{\"shareLicencePlate\": \"Yes\", \"shareParkingDuration\": \"Yes\",\"shareServiceUsages\": \"Yes\"}" https://localhost:8443/user/vishnu/privacy
  
curl -k -X PUT -d "{\"occupyTimeStamp\": \"09:15\", \"leaveTimeStamp\": \"12:15\", \"duration\": \"180\", \"parkingId\": \"A123\", \"usageServices\": \"Oil\"}" https://localhost:8443/user/vishnu/smartparking
  
curl -k -X PUT -d "{\"Occupied\": \"1\"}" https://localhost:8443/park/A123

curl https://localhost:8443/user/vishnu

curl -k -X DELETE https://localhost:8443/user/vishnu

cd go
VKONEPAL-M-W3Q8:go VKONEPAL$ pwd
/Users/VKONEPAL/go
VKONEPAL-M-W3Q8:go VKONEPAL$ export GOPATH=/Users/VKONEPAL/go
VKONEPAL-M-W3Q8:go VKONEPAL$ pwd
/Users/VKONEPAL/go
VKONEPAL-M-W3Q8:go VKONEPAL$ go get github.com/streadway/amqp

curl -k -X POST -d "{\"username\": \"vishnu\", \"email\": \"vishnu@cisco.com\", \"password\": \"vishnu123\"}" https://128.107.1.74:8443/user
{"message":"User","status":"Added","username":"vishnu"}

curl -k -X PUT -d "{\"billingContact\": \"Vishnu Konepalli\", \"email\": \"vishnu@cisco.com\", \"address\": \"3550 Cisco Way San Jose CA 95134\", \"zipCode\": \"95134\", \"carLicensePlat\": \"5jdg099\"}" https://128.107.1.74:8443/user/vishnu/profile
{"message":"Profile Inserted","messageCode":"200","status":"OK"}

curl -k -X POST -d "{\"Parkingid\": \"A123\"}" https://128.107.1.74:8443/user/vishnu/park
{"message":"Parking Info Saved","messageCode":"200","status":"OK"}

curl -k -X PUT -d "{\"shareLicencePlate\": \"Yes\", \"shareParkingDuration\": \"Yes\",\"shareServiceUsages\": \"Yes\"}" https://128.107.1.74:8443/user/vishnu/privacy
{"message":"Privacy Info Saved","messageCode":"200","status":"OK"}

curl -k -X PUT -d "{\"occupyTimeStamp\": \"09:15\", \"leaveTimeStamp\": \"12:15\", \"duration\": \"180\", \"parkingId\": \"A123\", \"usageServices\": \"Oil\"}" https://128.107.1.74:8443/user/vishnu/smartparking
{"message":"Parking Info Saved","messageCode":"200","status":"OK"}

curl -k -X PUT -d "{\"Occupied\": \"1\"}" https://128.107.1.74:8443/park/A123
{"message":"Sensor Info Saved","messageCode":"200","status":"OK"}

curl https://128.107.1.74:8443/user/vishnu
{"address":"3550 Cisco Way San Jose CA 95134","billingContact":"Vishnu Konepalli","carLicensePlat":"5jdg099","duration":"180","email":"vishnu@cisco.com","leaveTimeStamp":"12:15","occupyTimeStamp":"09:15","parkingId":"A123","shareLicencePlate":"Yes","shareParkingDuration":"Yes","shareServiceUsages":"Yes","usageServices":"Oil","zipcode":"95134"}

curl -k -X DELETE https://128.107.1.74:8443/user/vishnu
{"message":"User Deleted","messageCode":"200","status":"OK"}


test.go
package main







import(
	"github.com/kellydunn/golang-geo"
        "fmt"
)

func main() {
	// Make a few points
	p := geo.NewPoint(42.25, 120.2)
	p2 := geo.NewPoint(42.25, 120.2)

	// find the great circle distance between them
	dist := p.GreatCircleDistance(p2)
	fmt.Println(dist)
}


Vendors/Vendors.go
// Copyright Cisco Systems, Inc.
// All Rights Reserved.
// Author Vishnu Konepalli @ vkonepal@cisco.com
package main

import (
	"fmt"
	"log"
	"net/http"
	"strings"
	"encoding/json"
	"io/ioutil"
	"log/syslog"
	"database/sql"
	_ "github.com/go-sql-driver/mysql"
)

func checkErr1(err error) {
	if err != nil {
		logwritter1.Notice("Error connecting to database")
		panic(err)
	}
}

func Vendors(w http.ResponseWriter, r *http.Request) {
	logwritter1.Notice("Processing PO info from Client IP Address: 128.107.1.74")
	fmt.Println("Processing PO info from Client IP Address: 128.107.1.74")
	var httpMethod = r.Method
	//fmt.Println(httpMethod)
	//RemoteIP := strings.Split(r.RemoteAddr, ":")
	//IPAddress := RemoteIP[0]
	//fmt.Println(IPAddress)
	if strings.EqualFold(httpMethod, "POST") {
		//var u Customer
		if r.Body == nil {
			logwritter1.Err("Unable to read HTTP Method")
			http.Error(w, "Please send a request body", 400)
			return
		}
		//err := json.NewDecoder(r.Body).Decode(&u)
		//body, err := ioutil.ReadAll(r.Body)
		if err != nil {
			logwritter1.Err("Unable to read HTTP Request Body")
			http.Error(w, err.Error(), 400)
			return
		}
		body, err := ioutil.ReadAll(r.Body)
		if err != nil {

		}
		//fmt.Println(string(body))

		var jsonBody map[string]string

		if err2 := json.Unmarshal(body, &jsonBody); err2 != nil {
			logwritter1.Err("Unable to read username from database")
			fmt.Print(err2.Error())
		}

		carLicensePlat := jsonBody["carLicensePlat"]
		parkingId := jsonBody["parkingId"]

		//fmt.Println(billingContact,address,email,zipcode,carLicensePlat)

		db, err := sql.Open("mysql", "vkonepal:cisco123@/ms")
		checkErr1(err)

		stmt, err := db.Prepare("INSERT purchaseorder SET carLicensePlat=?,parkingId=?")
		checkErr1(err)

		res, err := stmt.Exec(carLicensePlat, parkingId)
		checkErr1(err)

		res.LastInsertId()
		logwritter1.Notice("PO details for Licenseplate " + carLicensePlat + " Inserted")

		db.Close()

		var Vendors = map[string]string{}

		Vendors["status"] = "SUCCESS"
		Vendors["message"] = "PO info received and saved !!!"
		Vendors["messageCode"] = "200"

		//fmt.Println(Vendors)

		jsonInfo, err := json.Marshal(Vendors)
		if err != nil {
			logwritter1.Err("Error in PO JSON marchslling")
			fmt.Println("Error in PO JSON marchslling")
		}
		//fmt.Println(jsonInfo)
		//fmt.Println(Vendors)

		jsonInfo, err1 := json.Marshal(Vendors)

		//fmt.Println(jsonInfo)

		if err1 != nil {
			logwritter1.Err("Error in PO JSON marchslling")
			fmt.Println("Error in PO JSON marchslling")
		}

		w.Header().Add("Content-Type", "application/json")
		w.Write(jsonInfo)
		S := string(jsonInfo)
		fmt.Println(S)

		w.Write([]byte("Thank You"))

	}
}

var logwritter1, err = syslog.New(syslog.LOG_ERR, "Vendors")

func main() {

	fmt.Println("Vendors started on port 9443.....")

	//logwritter,err := syslog.New(syslog.LOG_ERR,"CMPE295B")
	defer logwritter1.Close()
	if err != nil {
		log.Fatal("Error in the System")
	} else {
		logwritter1.Notice("Starting Server")

		http.HandleFunc("/", Vendors)

		//error := http.ListenAndServeTLS(":9443", "/Users/VKONEPAL/IdeaProjects/vkr/server.crt", "/Users/VKONEPAL/IdeaProjects/vkr/server.key", nil)
		//error := http.ListenAndServeTLS(":8443", "/home/cloud-user/go/src/github.com/CMPE295B/server.crt", "/home/cloud-user/go/src/github.com/CMPE295B/server.key", nil)
		error := http.ListenAndServe(":9443", nil)
		logwritter1.Err("Unable to Start Vendors Server")
		fmt.Println("Vendors finished 456.....")
		if err != nil {
			logwritter1.Alert(error.Error())
		}
	}

}
